blog/content/post/2022-04-07-fobnail_attestation_phase.md

```diff
@@ -0,0 +1,173 @@
+           determine the integrity of the system. The purpose of this blog post
+           is to present the development progress of this project. This phase
+           was focused on attestation.'
+cover: /covers/usb_token.png
+author:
+    - artur.kowalski
+    - krystian.hebel
+layout: post
+published: true
+date: 2022-04-06
+archives: "2022"
+
+tags:
+  - fobnail
+  - firmware
+  - tpm
+  - attestation
+categories:
+  - Firmware
+  - Security
+
+---
+
+# About the Fobnail Token project
+
+The Fobnail Token is a project that aims to provide a reference architecture for
+building offline integrity measurement servers on the USB device and clients
+running in Dynamically Launched Measured Environments (DLME). It allows the
+Fobnail owner to verify the trustworthiness of the running system before
+performing any sensitive operation. This project was founded by [NlNet
+Foundation](https://nlnet.nl/). More information about the project can be found
+in the [Fobnail documentation](https://fobnail.3mdeb.com/). Also, make sure to
+read other posts related to this project by visiting
+[fobnail](https://www.blog.3mdeb.com/tags/fobnail/) tag.
+
+# Scope of current phase
+
+This phase was mostly about implementing attestation. This includes splitting
+current implementation of Fobnail (aka Verifier) code into separate modules.
+Module for previous phase (platform provisioning) is started only if Fobnail
+doesn't have platform-specific artifacts saved in flash. Module for attestation
+is always started, either as first stage or immediately after provisioning, to
+test if provisioning succeeded.
+
+We also fixed some hardware issues that forced us to not use nRF52. While it
+still can't be fully used due to inability to provision Fobnail Token itself,
+this moves us closer to final, hardware, remote (i.e. not running on platform
+being verified) implementation of Fobnail.
+
+# Attestation
+
+Attestation is a process during which Verifier (Fobnail Token) asks Attester
+(host platform) for Evidence (in our case this is signed TPM quote), and based
+on it creates Attestation Results. This process is additionally controlled by
+Appraisal Policy and Reference Values. In general, Attestation Results may have
+complex format that is later appraised by another party, but in case of Fobnail
+this is binary _good_/_bad_ output.
+
+![Diagram of Fobnail attestation flow](/img/Fobnail-flows-attestation.png)
+
+## Implementation
+
+Reference Values in form of RIM were created and passed to Fobnail Token in
+[previous phase](../2022-03-21-fobnail_3rd_phase). Per-platform Appraisal Policy
+is expected to be installed during remote platform provisioning. In case of
+local provisioning a default policy is used.
+
+Default policy for Fobnail includes comparison of hashes of PCRs 0-7 and 17-18,
+for SHA256 bank. It is checked by using `TPM2_Quote()` command in order to avoid
+sending true PCR values through potentially insecure channel during attestation,
```

Comment:
Are PCR values considered confidential?

Reply:
Confidential - no, but knowing them may open a way to replay attacks.

Reply:
So, we're doing that to provide strengthen device protection against reply attacks? Why not to say that straight?

Reply:
I reworded this paragraph, there are better protection mechanisms implicitly done by TPM.